proxy/proxy/client.go

```diff
@@ -359,9 +359,16 @@ func NewConnSrc(instance string, l net.Listener) <-chan Conn {
 // close. Returns an error if there are still active connections after waiting
 // for the whole length of the timeout.
 func (c *Client) Shutdown(termTimeout time.Duration) error {
-	termTime := time.Now().Add(termTimeout)
-	for termTime.After(time.Now()) && atomic.LoadUint64(&c.ConnectionsCounter) > 0 {
-		time.Sleep(time.Second)
+	term, tick := time.After(termTimeout), time.Tick(100 * time.Millisecond)
```

Comment:
nit: `defer tick.Stop()`. Even though this is (currently) on only the shutdown path, it seems prudent to not leak resources anyway.

Reply:
`time.Tick` returns a channel and can't be stopped, but I can use `NewTicker` instead.

Reply:
Whoops duh, thanks for figuring out what I meant.